util/src/main/java/io/kubernetes/client/util/generic/GenericKubernetesApi.java

```diff
@@ -824,7 +824,17 @@ private Call tweakCallForCoreV1Group(Call call) {
       return call;
     }
     HttpUrl url = call.request().url();
-    HttpUrl tweakedUrl = url.newBuilder().removePathSegment(1).setPathSegment(0, "api").build();
+    String basePath = customObjectsApi.getApiClient().getBasePath();
+    // count segments in the basePath
+    int offset = 0;
```

Comment:
I don't think this will work if there is a trailing `/` in basePath (e.g. `https://foo.bar/`) and I think it probably should, can you add a test case for that and validate

Reply:
Thanks for your review, actually the trailing `/` is removed in `ClientBuilder.build`, so basePath would not have a trailing `/` (unless the user create a `ApiClient` directly instead of using `ClientBuilder.build`)

https://github.com/kubernetes-client/java/blob/f20788272291c0e79a8c831d8d5a7dd94d96d2de/util/src/main/java/io/kubernetes/client/util/ClientBuilder.java#L432-L437